playground/streaming/benchmark.py

```diff
@@ -0,0 +1,551 @@
+            silent_duration=0.01,
+            skip_initial_load=True,
+        )
+
+    for transcriber_cycle_idx in tqdm(
+        range(args.transcriber_num_cycles), desc="Transcriber Cycles"
+    ):
+        for transcriber_name in tqdm(args.transcribers, desc="Transcribers"):
+            transcriber = get_transcriber(transcriber_name, input_device)
+            if not args.transcriber_use_mic:
+                input_device.load()
+            transcriber_task = transcriber.start()
+
+            if args.transcriber_use_mic:
+
+                async def record_audio_task():
+                    while True:
+                        chunk = await input_device.get_audio()
+                        transcriber.send_audio(chunk)
+
+                send_audio = asyncio.create_task(record_audio_task())
+            else:
+
+                async def send_audio_task():
+                    while not input_device.is_done():
+                        chunk = await input_device.get_audio()
+                        transcriber.send_audio(chunk)
+                        await asyncio.sleep(sleep_time)
+
+                send_audio = asyncio.create_task(send_audio_task())
+
+            # `get` from `transcriber.output_queue` until it's empty for 5 seconds
+            pbar = tqdm(
+                desc=f"{transcriber_name.title()} Transcribing",
+                total=input_device.duration if not args.transcriber_use_mic else None,
+                unit="chunk",
+            )
+            while True:
+                try:
+                    transcription = await asyncio.wait_for(
+                        transcriber.output_queue.get(), timeout=5
+                    )
+                    # update the progress bar status
+                    pbar.update(round(transcriber.audio_cursor - pbar.n, 2))
+                except asyncio.TimeoutError:
+                    logger.debug(
+                        f"[Transcriber: {transcriber_name}] Transcriber queue is empty, stopping transcription..."
+                    )
+                    send_audio.cancel()
+                    break
+            if not args.transcriber_use_mic:
+                pbar.update(pbar.total - pbar.n)
+            transcriber.terminate()
+
+
+async def main():
+    result_file_path = os.path.join(args.results_dir, args.results_file)
+    if not args.just_graphs:
+        if args.agents:
+            await run_agents()
+        if args.transcribers:
+            await run_transcribers()
+        if args.synthesizers:
+            await run_synthesizers()
+
+        trace_results = span_exporter.get_finished_spans()
+        final_spans = defaultdict(list)
+        for span in trace_results:
+            duration_ns = span.end_time - span.start_time
+            duration_s = duration_ns / 1e9
+            final_spans[span.name].append(duration_s)
+
+        scope_metrics = reader.get_metrics_data().resource_metrics[0].scope_metrics
+        if len(scope_metrics) > 0:
```

Comment:
this is the same as the code in the specificstatisticsreader no?

Reply:
Yes, its the same but with a few changes. Refactored it to a separate method in tracing_utils.py.